python/ray/dag/compiled_dag_node.py

```diff
@@ -1111,6 +1125,14 @@ def _add_edge(
                 # Add an edge from the writer to the reader.
                 _add_edge(graph, idx, downstream_idx)
                 if task.dag_node.type_hint.requires_nccl():
+                    if _is_same_actor(idx, downstream_idx):
+                        logger.error(
+                            "Detect a deadlock caused by using NCCL channels to "
+                            "transfer data between tasks on the same actor. Please "
+                            'remove `TorchTensorType(transport="nccl")` between '
```

Comment:
This is a workaround, right? We should mention that, "To work around the issue, please ...".

Also, does shared memory work? Should ``TorchTensorType()` be used?

Reply:
> Should ``TorchTensorType()` be used?

If both the reader and the writer are on the same actor, `IntraProcessChannel` will be used. However, if users specify `TorchTensorType`, NCCL channel will be created instead.